kmake.mk

```diff
@@ -659,7 +659,6 @@ $(DIST_FOLDER).tar.gz: COMP=z
 $(DIST_FOLDER).tar.bz2: COMP=z
 $(addprefix $(DIST_FOLDER).tar.,$(DIST_SUFFIXES)): dist
 	$(call printcmd,TAR,$@)
-	$(Q)rm -f $@
 	$(Q)tar -c$(COMP) -f $@ $(DIST_FOLDER)
```

Comment:
What if tar fails, then the old archive might be left over?

Reply:
Oh, you're right.  I assumed that `.DELETE_ON_ERROR` is the default behaviour, but according to https://www.gnu.org/software/make/manual/make.html#Errors it isn't:
> Usually when a recipe line fails, if it has changed the target file at all, the file is corrupted and cannot be used—or at least it is not completely updated. Yet the file’s time stamp says that it is now up to date, so the next time make runs, it will not try to update that file. The situation is just the same as when the shell is killed by a signal; see Interrupts. So generally the right thing to do is to delete the target file if the recipe fails after beginning to change the file. make will do this if .DELETE_ON_ERROR appears as a target. This is almost always what you want make to do, but it is not historical practice; so for compatibility, you must explicitly request it.

Thus, please do not merge this change.